uamqp/aio/_connection_async.py

```diff
@@ -451,11 +452,10 @@ async def listen(self, wait=False, batch=1, **kwargs):
                 )
                 return
             try:
-                tasks = [asyncio.ensure_future(self._listen_one_frame(**kwargs)) for _ in range(batch)]
-                await asyncio.gather(*tasks)
+                for _ in range(batch):
+                    await asyncio.ensure_future(self._listen_one_frame(**kwargs))
             except ValueError:
-                for task in tasks:
-                    task.cancel()
+                pass
```

Comment:
Should we be silently passing this?

Reply:
I have updated the code to align with the sync implementation (break instead of catching error by the returning value), perf is not impacted.